internal/app/machined/pkg/controllers/k8s/endpoint.go

```diff
@@ -276,7 +277,7 @@ func kubernetesEndpointWatcher(ctx context.Context, logger *zap.Logger, client *
 		}),
 	)
 
-	notifyCh := make(chan *corev1.Endpoints, 1) //nolint:staticcheck
+	notifyCh := make(chan *discoveryv1.EndpointSlice, 1)
 
 	informer := informerFactory.Core().V1().Endpoints().Informer()
```

Comment:
same here: can we/should we use a filter by label?

Reply:
and, as there is ipv4 and ipv6 slice, we need to take care of correctly updating the result

Reply:
I've added a note to that part, we need a separate handler logic for that in the future. As of now, as we discussed, I'm leaving Endpoints in the k8s. Also I've opened PR to K/K to add missing RBAC rules.

PR: https://github.com/kubernetes/kubernetes/pull/133057
Slack 🧵: https://kubernetes.slack.com/archives/C0EN96KUY/p1752845789653939